sdk/storage/storage-blob/src/BlobClient.ts

```diff
@@ -1462,13 +1480,50 @@ export class BlobClient extends StorageClient {
    */
   public async downloadToBuffer(
     buffer: Buffer,
-    offset: number,
+    offset?: number,
     count?: number,
-    options: BlobDownloadToBufferOptions = {}
-  ): Promise<Buffer> {
+    options?: BlobDownloadToBufferOptions
+  ): Promise<Buffer>;
+
+  public async downloadToBuffer(
+    param1: Buffer | number = 0,
+    param2: number = 0,
+    param3: BlobDownloadToBufferOptions | number = 0,
+    param4: BlobDownloadToBufferOptions = {}
+  ) {
+    let buffer: Buffer;
+    let offset: number;
+    let count: number;
+    let options = param1 instanceof Buffer ? param4 : (param3 as BlobDownloadToBufferOptions);
     const { span, spanOptions } = createSpan("BlobClient-downloadToBuffer", options.spanOptions);
 
     try {
+      if (param1 instanceof Buffer) {
+        buffer = param1;
+        offset = param2;
+        count = param3 as number;
+      } else {
+        offset = param1;
+        count = param2 as number;
+        try {
+          buffer = Buffer.alloc(
+            count > 0
+              ? count
+              : (await this.getProperties({
+                  ...options,
+                  spanOptions
+                })).contentLength! - offset
+          );
+        } catch (error) {
+          throw new Error(
+            `Unable to allocate the buffer of size = count or (blobContentLength - offset) (in bytes). 
```

Comment:
IIUC, if you try to download a blob that doesn't exist, you'll get an error that complains about `buffer`. Perhaps this error should only be thrown when `buffer` can't be allocated, but errors from `getProperties` should be propagated as-is?

Reply:
Agree, we should only handle the allocation error (are there multiple kinds?); otherwise not handling at all seems better.

Reply:
There may be multiple kinds, but I think if we just move the getProperties call out of the try/catch block and let any of its errors bubble out, we'll be in an ok place.

Reply:
Agree.

After moving getProperties call out of try catch block, the error message can log the `count` number in the message too.

Reply:
Addressed.